src/metabase/models/dashboard.clj

```diff
@@ -258,6 +265,29 @@
                     (= prev-card-ids new-card-ids))        (deferred-tru "rearranged the cards")
                :else                                       (deferred-tru "modified the cards"))))
 
+         (when (or (:tabs changes) (:tabs removals))
+           (let [prev-tabs     (:tabs prev-dashboard)
+                 new-tabs      (:tabs dashboard)
+                 prev-tab-ids  (set (map :id prev-tabs))
+                 num-prev-tabs (count prev-tab-ids)
+                 new-tab-ids   (set (map :id new-tabs))
+                 num-new-tabs  (count new-tab-ids)
+                 num-tabs-diff (abs (- num-prev-tabs num-new-tabs))]
+             (cond
+               (and
+                 (set/subset? prev-tab-ids new-tab-ids)
+                 (< num-prev-tabs num-new-tabs))         (deferred-trun "added a tab" "added {0} tabs" num-tabs-diff)
+
+               (and
+                 (set/subset? new-tab-ids prev-tab-ids)
+                 (> num-prev-tabs num-new-tabs))         (deferred-trun "removed a tab" "removed {0} tabs" num-tabs-diff)
+
+               (and (= num-prev-tabs num-new-tabs)
+                    (= prev-tab-ids new-tab-ids)
+                    (= (set (map :name prev-tabs))
+                       (set (map :name new-tabs))))      (deferred-tru "rearranged the tabs")
```

Comment:
I think this is better expressed as "the set of tabs is the same excluding their positions"
```suggestion
               (= (set (map #(dissoc % :position) prev-tabs)
                  (set (map #(dissoc % :position) new-tabs))      (deferred-tru "rearranged the tabs")
```

Reply:
Nice, I updated it here 6ba8e0b6ad